osu.Game/Screens/OnlinePlay/Match/RoomSubScreen.cs

```diff
@@ -43,18 +45,27 @@ public abstract class RoomSubScreen : OnlinePlaySubScreen, IPreviewTrackOwner
         [Cached]
         protected OnlinePlayBeatmapAvailablilityTracker BeatmapAvailablilityTracker { get; }
 
+        private readonly Container content = new Container { RelativeSizeAxes = Axes.Both };
+
         protected RoomSubScreen()
         {
             BeatmapAvailablilityTracker = new OnlinePlayBeatmapAvailablilityTracker
             {
-                SelectedItem = { BindTarget = SelectedItem },
+                SelectedItem = { BindTarget = SelectedItem }
             };
+
+            base.AddInternal(content);
         }
 
+        // This is a bit ugly but we don't have the concept of InternalContent so it'll have to do for now. (https://github.com/ppy/osu-framework/issues/1690)
+        protected override void AddInternal(Drawable drawable) => content.Add(drawable);
+        protected override bool RemoveInternal(Drawable drawable) => content.Remove(drawable);
+        protected override void ClearInternal(bool disposeChildren = true) => content.Clear(disposeChildren);
+
         [BackgroundDependencyLoader]
         private void load(AudioManager audio)
         {
-            AddInternal(BeatmapAvailablilityTracker);
+            base.AddInternal(BeatmapAvailablilityTracker);
```

Comment:
i don't understand what is going on here.

Reply:
coming from https://github.com/ppy/osu/pull/11516#discussion_r559216271's discussion.

basically the beatmap tracker component may be removed by multiplayer or playlist screen via using `InternalChildren` instead of `AddInternal`, wasn't sure what the right way to fix is, so I went with "forwarding internal management".

Reply:
would help if you don't resolve comment threads and leave that to the person that opened them (for visibility).